libzfs/src/lib.rs

```diff
@@ -157,15 +166,23 @@ pub fn enumerate_vdev_tree(tree: &nvpair::NvList) -> io::Result<VDev> {
     match x {
         ref x if x == sys::VDEV_TYPE_DISK => {
             let path = tree.lookup_string(utf8_to_str(sys::ZPOOL_CONFIG_PATH))?;
-            let devid = tree.lookup_string(utf8_to_str(sys::ZPOOL_CONFIG_DEVID))?;
-            let phys_path = tree.lookup_string(utf8_to_str(sys::ZPOOL_CONFIG_PHYS_PATH))?;
-            let whole_disk = tree.lookup_uint64(utf8_to_str(sys::ZPOOL_CONFIG_WHOLE_DISK))? == 1;
+            let devid = tree.lookup_string(utf8_to_str(sys::ZPOOL_CONFIG_DEVID))
+                .ok();
+            let phys_path = tree.lookup_string(utf8_to_str(sys::ZPOOL_CONFIG_PHYS_PATH))
+                .ok();
+            let is_log = tree.lookup_uint64(utf8_to_str(sys::ZPOOL_CONFIG_IS_LOG))
+                .map(|x| x == 1)
```

Comment:
order of map and ok functions reversed for is_log compared with whole_disk, is this intended?

Reply:
It doesn't matter really.

Both [Result](https://doc.rust-lang.org/std/result/enum.Result.html#method.map) and [Option](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.map) implement `map`.

In the first case it's mapping over the `Ok` variant, in the second case it's mapping over the `Some` variant.